repo/maintenancestats/types_test.go

```diff
@@ -0,0 +1,102 @@
+package maintenancestats
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+type unmarshalable struct {
+	Data    string
+	Channel chan int
+}
+
+func (u *unmarshalable) Kind() string {
+	return u.Data
+}
+
+func TestBuildExtra(t *testing.T) {
+	um := unmarshalable{
+		Data: "fake",
+	}
+
+	cases := []struct {
+		name        string
+		stats       Kind
+		expected    Extra
+		expectedErr string
+	}{
+		{
+			name:        "nil stats",
+			expectedErr: "invalid stats",
+		},
+		{
+			name:        "marshal fails",
+			stats:       &um,
+			expectedErr: "error marshaling stats &{fake <nil>}: json: unsupported type: chan int",
+		},
+		{
+			name: "succeed",
+			stats: &CleanupMarkersStats{
+				DeletedEpochMarkerBlobs:       10,
+				DeletedDeletionWaterMarkBlobs: 20,
+			},
+			expected: Extra{
+				Kind: "cleanupMarkersStats",
+				Data: []byte("{\"deletedEpochMarkerBlobs\":10,\"deletedDeletionWaterMarkBlobs\":20}"),
```

Comment:
nit: you can use a raw string here, with ``, to avoid the escaping and increase visibility

Reply:
Done